***2. Задание:*** 

Имеется файл с различными оценками асессоров.

Формат файла: login tuid docid jud cjud.   

Пояснение к формату: login — логин асессора;   
uid — id асессора (user id);   
docid — id оцениваемого документа (document id);   
jud — оценка асессора (judgement);    
cjud — правильная оценка (correct judgement);   
разделитель — табуляция \t.  

Оценки могут принимать значение [0, 1], т.е. задание, которое сделали асессоры, имеет бинарную шкалу.    

Используя данные об оценках, установите, какие асессоры хуже всего справились с заданием. На какие показатели вы ориентировались и какие метрики вы использовали для ответа на этот вопрос? Можно ли предложить какие-то новые метрики для подсчета качества асессоров с учетом природы оценок у этого бинарного задания?     

Опишите подробно все этапы вашего решения.

In [1]:
import pandas as pd
import matplotlib

%pylab inline
%matplotlib inline

Populating the interactive namespace from numpy and matplotlib


In [2]:
df = pd.read_csv('/Users/kateosti/Downloads/data_task3.csv', sep='\t', index_col=False)

In [3]:
df.dtypes

login    object
uid       int64
docid     int64
jud       int64
cjud      int64
dtype: object

In [4]:
df.shape

(250000, 5)

In [5]:
df.head(5)

login  uid  docid  jud  cjud
0  assessor158  158      0    0     0
1  assessor238  238      0    0     0
2  assessor488  488      0    0     0
3  assessor136  136      0    0     0
4  assessor300  300      0    0     0

In [7]:
# Проверим ли все логины уникальны
df.login.nunique()

600

In [8]:
b = df.login.unique()

In [9]:
len(b)

600

Вывод: логины уникальны и не повторяются.

In [10]:
# Посмотрим сколько не правильно оценнных заданий ассесеров , сравнив с корректной оценкой системы
df.query('jud == 0 & cjud == 0')

login  uid  docid  jud  cjud
0       assessor158  158      0    0     0
1       assessor238  238      0    0     0
2       assessor488  488      0    0     0
3       assessor136  136      0    0     0
4       assessor300  300      0    0     0
...             ...  ...    ...  ...   ...
249989  assessor145  145  49997    0     0
249990  assessor397  397  49998    0     0
249991  assessor478  478  49998    0     0
249992  assessor564  564  49998    0     0
249993  assessor423  423  49998    0     0

[185406 rows x 5 columns]

***Вывод:***   Все эти логины ассесоров, сделали не правильно оценку заданий.

In [11]:
# Посмотрим сколько правильно оценнных заданий ассесеров , сравнив с корректной оценкой системы

df.query('jud == 1 & cjud == 1')

login  uid  docid  jud  cjud
5       assessor123  123      1    1     1
7       assessor409  409      1    1     1
8       assessor396  396      1    1     1
9       assessor295  295      1    1     1
330     assessor144  144     66    1     1
...             ...  ...    ...  ...   ...
249995  assessor208  208  49999    1     1
249996  assessor139  139  49999    1     1
249997  assessor333  333  49999    1     1
249998  assessor160  160  49999    1     1
249999  assessor175  175  49999    1     1

[24916 rows x 5 columns]

1. ***Вывод:*** получилось что 185406 ассесоров не правильно выполнили задание и получили оценку 0,
а 24916 ассесоров сделали правильно.

2. На какие показатели вы ориентировались и какие метрики вы использовали для ответа на этот вопрос?

Я ориентировалась на оценку системы, 0 значит плохо, не засчитано. Использовала сравнение, оценку ассесора сравнила с оценкой системы.

3. Можно ли предложить какие-то новые метрики для подсчета качества асессоров с учетом природы оценок у этого бинарного задания?
Можно раздробить документ и оценивать его микрозаданиям. 
Например, соотвествует ли картинка подписи: да - 1, 0 - нет. 
Корректны ли ссылки да - 1, нет - 0.
Четкая ли картинка: да - 1, нет - 0.